### PR TITLE
Corrected json serialisation

### PR DESCRIPTION
**Pull request by jesper-friis**
Corrected JSON serialisation, JSON strings are UTF-8-encoded.

Please also make the JSON serialisation readable. Use two space indentation.

### CODE REVIEW

File: oteapi_optimade/dlite/entities/OPTIMADEStructureAttributes.yaml [hunk 1]
```diff
@@ -42,7 +42,7 @@ properties:
     type: float
     shape: [dimensionality, dimensionality]
     unit: Ångström
-    description: The three lattice vectors in Cartesian coordinates, in ångström (Å).
+    description: The three lattice vectors in Cartesian coordinates, in Ångström (Å).
```
Comment (CasperWA) [inline oteapi_optimade/dlite/entities/OPTIMADEStructureAttributes.yaml:45]: This description is lifted (and meant to be equal to) the OPTIMADE specification: https://github.com/Materials-Consortia/OPTIMADE/blob/develop/optimade.rst#lattice-vectors
```suggestion
    description: The three lattice vectors in Cartesian coordinates, in ångström (Å).
```

**Review by CasperWA — CHANGES_REQUESTED** [review]
To be honest. The JSON files here are just because this wouldn't work natively if the entities were in YAML, which I'd much prefer. Hence, the JSON files are not the "source".
These JSON files _are_ already utf-8 compliant, they have been generated thus:

First, ensuring one is in the repository root.
```json
import json
from pathlib import Path

import yaml

entity_filename = "OPTIMADEStructureAttributes"

yaml_entity = Path(f"oteapi_optimade/dlite/entities/{entity_filename}.yaml").resolve()

parsed_entity = yaml.safe_load(yaml_entity.read_text(encoding="utf-8"))

json_entity = yaml_entity.with_suffix(".json")
json_entity.write_text(json.dumps(parsed_entity), encoding="utf-8")
```

The byte characters you see are exactly UNICODE (utf-8), but I'd expect it to be manageable by any parser, as well as more safe for storing entities in any DB backend.

---

But changing the unit from Ångström to Å makes sense to me if pint will understand this 👍

**Comment by jesper-friis** [comment]
> To be honest. The JSON files here are just because this wouldn't work natively if the entities were in YAML, which I'd much prefer. 

DLite should not have problems with data models in YAML, you just have to provide them correctly to `dlite.storage_path`. Since we have standardised on JSON, dlite supports just appending a directory path with the JSON files to `dlite.storage_path`. For YAML you need to append it like "yaml:/some/path/*.yaml"

**Comment by CasperWA** [comment]
> > The byte characters you see are exactly UNICODE (utf-8), but I'd expect it to be manageable by any parser, as well as more safe for storing entities in any DB backend.
> 
> If you look at the diff, you see that the origin was the ASCII sequence "\u00c5ngstr\u00f6m". True, it is valid UTF-8, but is not in anyway understandable by neither humans or Pint.

This is Unicode, just escaped - not ASCII. See https://www.ascii-code.com/character/%C3%A5

**Comment by CasperWA** [comment]
> > To be honest. The JSON files here are just because this wouldn't work natively if the entities were in YAML, which I'd much prefer.
> 
> DLite should not have problems with data models in YAML, you just have to provide them correctly to `dlite.storage_path`. Since we have standardised on JSON, dlite supports just appending a directory path with the JSON files to `dlite.storage_path`. For YAML you need to append it like "yaml:/some/path/*.yaml".
> 
> But it is much preferable to follow the standard way to work with DLite than inventing your own

It is much preferable to have a choice that is more easily available and usable than having to conform to outdated conventions.

**Comment by jesper-friis** [comment]
> > > The byte characters you see are exactly UNICODE (utf-8), but I'd expect it to be manageable by any parser, as well as more safe for storing entities in any DB backend.
> > 
> > 
> > If you look at the diff, you see that the origin was the ASCII sequence "\u00c5ngstr\u00f6m". True, it is valid UTF-8, but is not in anyway understandable by neither humans or Pint.
> 
> This is Unicode, just escaped - not ASCII. See https://www.ascii-code.com/character/%C3%A5

Unicode "Å" is a sequence of 2-bytes (0x00, 0xc5). It is obviously not the same thing as the 6 byte sequence "\u00c5". The first is understood by Pint, the latter is not.

**Comment by CasperWA** [comment]
No longer relevant - fixes implemented instead in #169 and #170

**Pull request closed without merging**